sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -665,6 +675,27 @@ def __init__(self, windowing):
     def expand(self, pcoll):
       return pcoll | self._window_transform
 
+    @staticmethod
+    def _parse_time_unit(value, name):
+      time_units = {'s': 1, 'm': 60, 'h': 60 * 60, 'd': 60 * 60 * 12}
+      value, suffix = str(value)[:-1], str(value)[-1]
+      # Default to seconds if time unit suffix is not defined
+      if suffix.isnumeric():
+        value, suffix = value + suffix, 's'
+      elif not value or not value.isnumeric():
```

Comment:
This won't accept things like 1.5s. 

What if instead you broke the suffix off with a regex (e.g. `^(.*?)([^\d]*)$`) and then detected whether the suffix was in our time units dict. (This would allow multi-letter suffixes, like ms, or reject them (rather than giving errors like "s is a valid suffix, but 10hour is not a valid floating point number" even if we didn't support these).

Reply:
Good idea